### PR TITLE
Add GH actions to assemble a multiarch manifest for amd64 rocks

### PR DESCRIPTION
**Pull request by addyess**
This PR adds a gh action job which will build each rocks in this repository for each architecture it supports (currently only amd64) and generates a multiarch manifest for each rock annotated with the supported architecture.

The images are pushed on every build, but the  multiarch manifests are pubilshed to ghcr.io on merges to main

### CODE REVIEW

**Review by berkayoz — COMMENTED** [review]
Looks good overall, I see trivy detection is running but we are missing information under the Security tab is there something we need to do more? I also would love to listen to a small explanation on how the caching works in the workflow.

**Comment by addyess** [comment]
TLDR on caching:  github-actions allows you to cache anything from the worker machine, and import that cache later.  The strategy here is to store into the cache on push to `main`.  When the push occurs, we build the rocks and store the build state from lxd into the cache.  The next branch that comes along derived from `main`, can restore that cache.  If one rock changes, but the other does not -- we get the benefit of the cache to not wait for the rebuild of the unchanged one

The key to the cache is based off the name of the rock and  filehash of the directory containing the rockcraft.yaml

All of this was inheritted from the [operator-workflow](https://github.com/canonical/operator-workflows) version of `build-rocks` since i couldn't get the merges in there. 

I imagine as a TODO -- we'd like these actions to become a workflow that can be reused on other k8s projects (dns, localpv, etc...)

**Review by berkayoz — APPROVED** [review]
LGTM, we might want to figure out why Trivy results are not showing up under the Security tab but that's already out of scope for this card.

One small comment we can address later, it might be nice to have an `1.14.5` tag that points to the latest `1.14.5-ck*` image?
